keras_nlp/src/utils/transformers/convert_albert.py

```diff
@@ -0,0 +1,195 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+
+from keras_nlp.src.utils.preset_utils import HF_CONFIG_FILE
+from keras_nlp.src.utils.preset_utils import get_file
+from keras_nlp.src.utils.preset_utils import jax_memory_cleanup
+from keras_nlp.src.utils.preset_utils import load_config
+from keras_nlp.src.utils.transformers.safetensor_utils import SafetensorLoader
+
+
+def convert_backbone_config(transformers_config):
+    return {
+        "vocabulary_size": transformers_config["vocab_size"],
+        "num_layers": transformers_config["num_hidden_layers"],
+        "num_heads": transformers_config["num_attention_heads"],
+        "embedding_dim": transformers_config["embedding_size"],
+        "hidden_dim": transformers_config["hidden_size"],
+        "intermediate_dim": transformers_config["intermediate_size"],
+        "num_groups": transformers_config["num_hidden_groups"],
+        "num_inner_repetitions": transformers_config["inner_group_num"],
+        "dropout": transformers_config["attention_probs_dropout_prob"],
+        "max_sequence_length": transformers_config["max_position_embeddings"],
+        "num_segments": transformers_config["type_vocab_size"],
+    }
+
+
+def convert_weights(backbone, loader):
+    # Embeddings
+    loader.port_weight(
+        keras_variable=backbone.token_embedding.embeddings,
+        hf_weight_key="albert.embeddings.word_embeddings.weight",
+    )
+    loader.port_weight(
+        keras_variable=backbone.position_embedding.position_embeddings,
+        hf_weight_key="albert.embeddings.position_embeddings.weight",
+    )
+    loader.port_weight(
+        keras_variable=backbone.segment_embedding.embeddings,
+        hf_weight_key="albert.embeddings.token_type_embeddings.weight",
+    )
+
+    # Normalization
+    loader.port_weight(
+        keras_variable=backbone.embeddings_layer_norm.gamma,
+        hf_weight_key="albert.embeddings.LayerNorm.weight",
+    )
+    loader.port_weight(
+        keras_variable=backbone.embeddings_layer_norm.beta,
+        hf_weight_key="albert.embeddings.LayerNorm.bias",
+    )
+
+    # Encoder
+    loader.port_weight(
+        keras_variable=backbone.embeddings_projection.kernel,
+        hf_weight_key="albert.encoder.embedding_hidden_mapping_in.weight",
+        hook_fn=lambda hf_tensor, _: np.transpose(hf_tensor, axes=(1, 0)),
+    )
+    loader.port_weight(
+        keras_variable=backbone.embeddings_projection.bias,
+        hf_weight_key="albert.encoder.embedding_hidden_mapping_in.bias",
+    )
+
+    keras_prefix = backbone.get_layer("group_0_inner_layer_0")
+    hf_prefix = "albert.encoder.albert_layer_groups.0.albert_layers.0."
```

Comment:
Also, this seems two hardcoded. Both Keras and HF have config support for multiple groups, but this assumes that `num_hidden_groups=1` and maybe that `inner_group_num=1` too.

If it's easy to write this in such a way that we are not assuming those constants we should just do that. Otherwise, we should at least throw if `num_hidden_groups > 1` or `inner_group_num > 1`, which could be true for a user upload.

Reply:
Yeah. Makes sense. I have made changes to fix it. Does this approach works?